src/main/java/teammates/ui/webapi/output/SessionResultsData.java

```diff
@@ -352,7 +352,10 @@ public CommentOutput getParticipantComment() {
 
         public CommentOutput(FeedbackResponseCommentAttributes frc, String commentGiverName, String lastEditorName) {
             super(frc);
-            this.commentGiverName = commentGiverName;
+
+            // If response is anonymous to everyone, then comment giver's name is also hidden.
+            this.commentGiverName = (frc.showGiverNameTo.isEmpty()) ? null : commentGiverName;
```

Comment:
There are multiple things here:
- You should hide not only the giver name but also the `giver` itself.
- You should also hide lastEditorName and lastEditor
- We only want to do so for `participantComment`. I think you can just blindly hiding the field without looking at the visibility for participant comment as the fields should not be used in the frontend.
- If you want to follow the visibility, it should be the comment visibility not the response visibility.

 Maybe create a method here and call it when `participantComment` get constructed?

Reply:
Updated to hide `commentGiver`, `commentGiverName`, `lastEditorEmail`, `lastEditorName` for participantComment.